src/overlays/actors/ovl_En_Bubble/z_en_bubble.c

```diff
@@ -32,73 +31,399 @@ const ActorInit En_Bubble_InitVars = {
+    ColliderJntSphElementDim* dim;
+    Vec3f src;
+    Vec3f dest;
+
+    dim = &this->colliderSphere.elements[0].dim;
+    src.x = dim->modelSphere.center.x;
+    src.y = dim->modelSphere.center.y;
+    src.z = dim->modelSphere.center.z;
+
+    Matrix_MultVec3f(&src, &dest);
+    dim->worldSphere.center.x = dest.x;
+    dim->worldSphere.center.y = dest.y;
+    dim->worldSphere.center.z = dest.z;
+    dim->worldSphere.radius = dim->modelSphere.radius * (1.0f + this->expansionWidth);
+    this->colliderSphere.elements[1].dim = *dim;
+}
+
+void EnBubble_Init(Actor* thisx, PlayState* play) {
+    EnBubble* this = THIS;
+    s32 pad;
+
+    ActorShape_Init(&this->actor.shape, 16.0f, ActorShadow_DrawCircle, 0.2f);
+    Collider_InitJntSph(play, &this->colliderSphere);
+    Collider_SetJntSph(play, &this->colliderSphere, &this->actor, &sJntSphInit, this->colliderElements);
+    CollisionCheck_SetInfo2(&this->actor.colChkInfo, DamageTable_Get(9), &sColChkInfoInit);
+    this->actor.hintId = 0x16;
+    this->bounceDirection.x = Rand_ZeroOne();
+    this->bounceDirection.y = Rand_ZeroOne();
+    this->bounceDirection.z = Rand_ZeroOne();
+    EnBubble_Vec3fNormalize(&this->bounceDirection);
+    this->velocityFromBounce.x = this->bounceDirection.x * 3.0f;
+    this->velocityFromBounce.y = this->bounceDirection.y * 3.0f;
+    this->velocityFromBounce.z = this->bounceDirection.z * 3.0f;
+    EnBubble_SetDimensions(this, 0);
+    this->actionFunc = EnBubble_Wait;
+}
+
+void EnBubble_Destroy(Actor* thisx, PlayState* play) {
+    EnBubble* this = (EnBubble*)thisx;
+
+    Collider_DestroyJntSph(play, &this->colliderSphere);
+}
+
+void EnBubble_Wait(EnBubble* this, PlayState* play) {
+    if (EnBubble_DetectPop(this, play) != 0) {
+        this->explosionCountdown = func_8089F59C(this);
+        this->actionFunc = EnBubble_Pop;
+    } else {
+        EnBubble_Fly(this, play);
+        this->actor.shape.yOffset = (this->expansionHeight + 1.0f) * 16.0f;
+        CollisionCheck_SetAC(play, &play->colChkCtx, &this->colliderSphere.base);
+        CollisionCheck_SetOC(play, &play->colChkCtx, &this->colliderSphere.base);
+    }
+}
+
+void EnBubble_Pop(EnBubble* this, PlayState* play) {
+    if (EnBubble_Explosion(this, play) >= 0) {
+        SoundSource_PlaySfxAtFixedWorldPos(play, &this->actor.world.pos, 60, NA_SE_EN_AWA_BREAK);
+        Actor_MarkForDeath(&this->actor);
+    }
+}
+
+void EnBubble_Disappear(EnBubble* this, PlayState* play) {
+    s32 temp_v0 = func_8089F8BC(this);
+
+    if (temp_v0 >= 0) {
+        this->actor.shape.shadowDraw = ActorShadow_DrawCircle;
+        this->explosionCountdown = temp_v0;
+        this->actionFunc = EnBubble_Regrow;
+    }
+}
+
+void EnBubble_Regrow(EnBubble* this, PlayState* play) {
+    if (func_8089F908(this)) {
+        this->actionFunc = EnBubble_Wait;
+    }
+    CollisionCheck_SetAC(play, &play->colChkCtx, &this->colliderSphere.base);
+    CollisionCheck_SetOC(play, &play->colChkCtx, &this->colliderSphere.base);
+}
```

Comment:
Haaaang on a minute, if I search for `EnBubble_Disappear` I see it's checked for in the draw function but it never actually gets set as an actionfunc, and therefore neither does Regrow.

Is that right? Unused code? Did OOT use it?

Reply:
OoT did not use it either, appears to be unused there as well